runtime/lua/vim/shared.lua

```diff
@@ -188,6 +188,31 @@ local function pesc(s)
   return s:gsub('[%(%)%.%%%+%-%*%?%[%]%^%$]', '%%%1')
 end
 
+--- Test if `prefix` is a prefix of `s`
+--
+-- @tparam string s String to check
+-- @tparam string prefix Potential prefix
+-- @treturn boolean True if prefix is a prefix of s
+local function startswith(s, prefix)
+  assert(type(s) == 'string', 'Input must be a string')
+  assert(type(prefix) == 'string', 'Prefix must be a string')
+	return s:sub(1, #prefix) == prefix
+end
+
+--- Test if `suffix` is a suffix of `s`
+--
+-- @tparam string s String to check
+-- @tparam string suffix Potential suffix
+-- @treturn boolean True if suffix is a suffix of s
+local function endswith(s, suffix)
+  assert(type(s) == 'string', 'Input must be a string')
+  assert(type(suffix) == 'string', 'Suffix must be a string')
+	return #suffix == 0 or s:sub(-#suffix) == suffix
+end
+
+string.startswith = startswith
+string.endswith = endswith
```

Comment:
what's this? We don't want want to monkey-patch Lua if we can avoid it. Keep stuff on `vim` module in most cases.